server/controllers/orderItemController.js

```diff
@@ -0,0 +1,49 @@
+const OrderItemModel = require('../Models/orderItem');
+const PizzaModel = require('../Models/pizza');
+
+module.exports = {
+    createOrder : async (req, res) => { 
+        PizzaModel.findOne({ _id : req.body.pizza_id })
+            .exec()
+            .then(pizza => {
+                let orderItem = new OrderItemModel({
+                    pizza_id: req.body.pizza_id,
+                    quantity: req.body.quantity,
+                    totalAmount: pizza.price * req.body.quantity
+                });
+                orderItem.save().then( response => {
+                    res.status(201).json({ Order: { name: pizza.name, price: pizza.price, quantity: response.quantity, totalAmount: response.totalAmount } });
+                }).catch(err => res.status(500).json({ error: err.message }));  
+            })
+            .catch(err => res.status(500).json({ error: 'The Pizza you want to order is inexistent' }))
+        // 
+        
+    },
+    //list of orders
+    getOrders: async(req, res) => {
+        OrderItemModel.find((err, orders) => {
+            if(err) res.status(400).json({ error : err.message })
+            res.status(200).json({ orders, length: orders.length });
+        });
+    },
+    // details of an individual order
+    getOrderDetails: (req, res) => {
+        OrderItemModel.findOne({_id : req.params.id})
+            .exec()
+            .then(order => {
+                PizzaModel.findOne({ _id: order.pizza_id })

```

Comment:
Is this because the order only have ID of the pizza, so to get details, you need to do this fetch?

Reply:
Yes, the order has the pizza id, quantity and amount. This fetch enables me get the name and price of the pizza order